File: app/src/warsh/java/com/quran/labs/androidquran/data/QuranDataModule.java
package com.quran.labs.androidquran.data;

import com.quran.data.source.PageProvider;
import com.quran.data.source.QuranDataSource;
import dagger.Module;
import dagger.Provides;
import java.util.Map;

@Module
public class QuranDataModule {

  @Provides
  static PageProvider provideQuranPageProvider(Map<String, PageProvider> providers) {
    return providers.get("warsh");
  }

  @Provides
  static QuranDataSource provideQuranDataSource(PageProvider pageProvider) {
    return pageProvider.getDataSource();
  }
}

File: app/src/warsh/java/com/quran/labs/androidquran/data/QuranDataModule.kt
package com.quran.labs.androidquran.data

import android.view.Display
import com.quran.data.page.provider.common.QuranPageProvider
import com.quran.data.page.provider.common.size.QuranPageSizeCalculatorProvider
import com.quran.data.source.DisplaySize
import dagger.Module
import dagger.Provides

@Module
object QuranDataModule {
  @JvmStatic @Provides fun provideQuranPageProvider() =
      QuranPageProvider.provideWarshPageProvider()

  @JvmStatic @Provides fun provideQuranPageSizeCalculator(displaySize: DisplaySize) =
      QuranPageSizeCalculatorProvider.provideMadaniPageSizeCalculator(displaySize)
}
